Lib/test/test_index.py

```diff
@@ -66,10 +66,10 @@ def __index__(self):
         direct_index = my_int.__index__()
         operator_index = operator.index(my_int)
         self.assertEqual(direct_index, 8)
-        self.assertEqual(operator_index, 7)
+        self.assertEqual(operator_index, 8)
```

Comment:
This is the key change: for a subclass of `int`, we take `__index__` to be the true value instead of the integer's value.